public/app/types/explore.ts

```diff
@@ -204,12 +204,10 @@ export interface ExploreItemState {
    */
   cache: Array<{ key: string; value: ExplorePanelData }>;
 
-  // properties below should be more generic if we add more providers
-  // see also: DataSourceWithLogsVolumeSupport
-  logsVolumeEnabled: boolean;
-  logsVolumeDataProvider?: Observable<DataQueryResponse>;
-  logsVolumeDataSubscription?: SubscriptionLike;
-  logsVolumeData?: DataQueryResponse;
+  /**
+   * Supplementary queries are additional queries used in Explore, e.g. for logs volume
+   */
+  supplementaryQueries: SupplementaryQueries;
```

Comment:
Do you think you would need multiple supplementary queries of the same type? I'm thinking if it'd make sense to turn this into a list instead of a map 🤔 Do you know what would be the UI for sample logs with mixed data sources?

@grafana/explore-squad The reason why I'm asking is also that we will definitely need some refactoring in the future on top of that to support supplementary queries in mixed data sources. If it was a list we would add `datasourceId` to `SupplementaryQuery` object and filter when running queries. We will probably merge the data for supplementary queries to have all in single visuzalization.

Reply:
while it may make sense i feel like it'd be better to tackle it in a separate task, given it will most likely require additional changes i wouldn't block this PR because of this

Reply:
Good question! Based on your suggestion, I agree that it would. But also, I'd like to keep this PR strictly related to refactor. And  all logic related to fixing logs volume/supporting queries for mixed data sources (e.g. adding data source id/merging data/...) would be done in separate PR (we already have issue for it https://github.com/grafana/grafana/issues/56314).

Reply:
To move on with this and to start to work on log samples, I will merge this with `supplementaryQueries` being map.  But we will definitely get back to it and we can re-evaluate as we need to fix log volume for mixed data sources.  Thank you for bringing this up @ifrost.